integrationtests/src/test/java/io/kroxylicious/proxy/KrpcFilterIT.java

```diff
@@ -203,9 +203,65 @@ public void shouldModifyProduceMessage(KafkaCluster cluster, Admin admin) throws
         }
     }
 
+    @Test
+    public void offsetManuallyManaged(KafkaCluster cluster, Admin admin) throws Exception {
```

Comment:
I don't understand the motivation for adding this test.  The ApiVersion/BrokerAddress filters don't care about offsets in any way, so why is this test important to have.

Reply:
I asked on the slack channel and it was suggested by @SamBarker, having some test for a manually managed offset.

Reply:
I am looking for a test that validates that kroxylicious is not messing with offset handling (i.e. normal operation everything works) as a safety net for future changes where it does manipulate the requests. 

I think there are several scenarios that need covered.
1. General usage is unaffected clients can manage offsets normally when kroxylicious *is not manipulating* the commit
2. A test that verifies what gets written to the physical cluster is identical to what was committed to the virtual cluster
3. A test that verifies what gets written to the physical cluster is different but equivalent to what was committed to the virtual cluster.
4. General usage is unaffected clients can manage offsets normally when kroxylicious *is manipulating* the commit 

Which when I enumerate them like that is really two themes. Tests validate Clients of the virtual cluster operate normally regardless of what the proxy is doing. Tests validate that Kroxylicious is transforming the message as intended by the filter set.

My take is that should be our standard "template" for each API or Flow we want to support (I'm not sure what level of abstraction makes sense to focus those around).

Reply:
> I am looking for a test that validates that kroxylicious is not messing with offset handling 

what makes you pick out offset handling for requiring extra tests?  why not say, aspects of the consumer group protocol?  If we have a offsetmaniluatingfilter then sure we'd want test cases around that but as there is nothing in kroxyilicious's core manipulates offsets, I'm failing to get your motivation.

Reply:
I'm looking for it any and everywhere 😁  I don't intend to be specific to offset handling, I was just wanting to establish the pattern here.

Reply:
I'm sorry. I still don't get it.  The filter chain under test is ApiVersions/BrokerAddress.  Kroxy isn't even going to decode the RPCs related to offsets. Kroxy is nothing more than a boring old byte pipe.  I don't understand the value that this test brings.

Reply:
In the short term validating that it is a boring pipe seems valuable to me. 

I don't understand we we wouldn't want test coverage that the pipe does what we expect, even if that is don't change it. 

IIRC we are/going to re-write clientids and groupIds which must be part of what is committed to Kafka, in multi tenant land. It seems valuable to me that should that handling go wrong we have a test demonstrating that the basics work.

Reply:
The MultiTenantITs are defintely going to cover offsets.  Those tests do that already (to a small extent), and we are in the process of expanding that to cover other offset use-cases.

Reply:
> Where would we stop? Would we test all right tests for every unfiltered RPC?
Yes. Baseline proof that the proxy doesn't break the individual RPC's. I do not understand how that is remotely controversial. I'm not advocating for in-depth coverage or testing every conceivable edge case just a basic smoke test that each RPC is proxied correctly. 

> The [MultiTenantITs](https://github.com/search?q=repo%3Akroxylicious%2Fkroxylicious%20consumeOneAndOffsetCommit&type=code) are definitely going to cover offsets (and client id and group ids). That's justified because it is intercepting those RPCs. Those tests already cover offsets (to a small extent), and we are in the process of growing that to cover other offset use-cases right now.

So when one of those breaks I think there is value in knowing that the basic pipe doesn't affect things.

Reply:
> So when one of those breaks I think there is value in knowing that the basic pipe doesn't affect things.

Totally agree with Sam here.

Reply:
My point is kroxy (in KrpcFilterIT configuration) is not decoding the RPCs used by the test cases you are proposing writing. No filters will be  firing. No extra code running.  All you are exercising is the kroxy's ability to shunt opaque bytes to/from the network (there's a correlation id rewrite, but that's non RPC specific common code).  The integration test is already demonstrating that shunting bytes works.   Testing it will n different RPCs brings us zero.

This to me is like saying a washing machine isn't tested until you've washed a sock of every colour.

Reply:
>Testing it will n different RPCs brings us zero.

Ok that's the fundamental disconnect. I'm saying that it brings us value. It might be small in general but when something breaks in a specific filter validating that the rest of the proxy is not causing issues.

A couple of possible ways to break the impasse:
 1.  Add a NoOp filter which touches every RPC so that we are exercising the decode encode logic
 2. Add a Filter which adds a header to every RPC to mark it as having gone through the proxy so that the proxy under test is actually doing something.
 
 
 >This to me is like saying a washing machine isn't tested until you've washed a sock of every colour.
 
 To use your analogy I think it's akin to saying that the washing machine isn't tested until it has washed a bed sheet, a towel, a t-shirt a sock, and a silk shirt. The wash process is consistent between them but we are validating they all survive the wash cycle.

Reply:
Demonstrating fidelity in the decode encode logic seems like a valuable test.  However, I don't think using the client/broker is the best way to achieve that.   Couldn't we write a test that does that without the kafka client/broker?  Either as a unit test where we just round-trip decode->encode, and/or an network integrated test that just uses a fake client to send all possible RPCs/versions and a fake server that validated what is received (and the reciprocal for responses).  Using code generation for the tests might be worth considering.

Reply:
With #243 merged we now have this coverage of Kroxylicious decoding/encoding/intercepting/proxying all RPCs at each supported apiVersion. Does this unblock this PR?

Reply:
In my mind, the work on #243 makes the case for closing this PR (without merge) stronger.

Reply:
@franvila are you happy that #243 covers everything from here or is there still milage in this PR? 

My expectation is that #243 has things covered and we should close this one.

Reply:
Agree. Thanks @robobario!